lib/sorbet-rails/model_rbi_formatter.rb

```diff
@@ -215,7 +215,20 @@ def populate_collection_assoc_getter_setter(assoc_name, reflection)
 
   def populate_generated_enum_methods
     @model_class.defined_enums.each do |enum_name, enum_hash|
-      @generated_class_sigs["#{enum_name.pluralize}"] = { ret: "T::Hash[T.any(String, Symbol), Integer]"}
+      # Rails documents these values as being integers, but in practice any
+      # type is allowed.
+      value_types = enum_hash.values.map {|x| x.class.to_s }.uniq
```

Comment:
```suggestion
      value_types = enum_hash.values.map {|x| x.class.name }.uniq
```

Reply:
Can we use `value_types` for the sig at line 228?

```
        @generated_class_sigs["#{enum_val}"] = {
          args: [ name: :args, arg_type: :rest, value_type: 'T.untyped' ],
          ret: @model_relation_class_name,
        }
```
(I think for this method, symbol & string are always acceptable as input, besides the value_types)